magpy/stream.py

```diff
@@ -27,15 +28,22 @@
 
 # Logging
 # ---------
-# Select the home directory of the user (platform independent)
-from os.path import expanduser
-home = expanduser("~")
+# Select the user's home directory (platform independent) or environment path
+if "MAGPY_LOG_PATH" in os.environ:
+    env_path = os.environ["MAGPY_LOG_PATH"]
+    if os.path.exists(env_path):
+        path_to_log = env_path
```

Comment:
this if/else is redundant

- remove the `env_path` variable (just use `path_to_log`)
- only have the if create the directories if the path doesn't exist (only os.makedirs is needed inside the if, but you need to invert the logic)